CMakeLists.txt

```diff
@@ -462,8 +469,8 @@ if(WIN32)
   option(ENABLE_DLL "Build a Windows DLL." ${BUILD_SHARED_LIBS})
   if(ENABLE_DLL)
     set(BUILD_DLL ON CACHE BOOL "")
-    add_definitions(-DDLL_NETCDF)
-    add_definitions(-DDLL_EXPORT)
+    add_compile_definitions(-DDLL_NETCDF)
+    add_compile_definitions(-DDLL_EXPORT)
```

Comment:
I am on the edge about `GenerateExportHeader`, haven't had enough experience with it. But note that using that would mean completely removing autotools.

Reply:
@LecrisUT I tried `target_compile_definitions ` but that caused several of the tests to fail. I forget which target now, but something was relying on this being global. It seems that some of the other targets also needed these when I'd tried to change them. I'll revisit this sometime this week and see if I can figure out what

Reply:
@ZedThree @K20shores, I just played around with `GenerateExportHeader`, and I don't recommend it because it generates the macros only for the compiler you are currently using [^1]. For internal compilation, that would be ok, but for example `DEPRECATED` would be useful to define in a portable way so that these can be added to to the exported header files.

[^1]: https://discourse.cmake.org/t/generateexportheader-and-multi-compiler-support/10102